src/libstore/store-api.cc

```diff
@@ -1099,7 +1126,7 @@ ref<Store> openStore(const std::string & uri_,
         auto parsedUri = parseURL(uri_);
         params.insert(parsedUri.query.begin(), parsedUri.query.end());
 
-        auto baseURI = parsedUri.authority.value_or("") + parsedUri.path;
+        auto baseURI = extractConnStr(parsedUri.authority.value_or("") + parsedUri.path);
```

Comment:
Shouldn't `extractConnStr` be applied in `SSHStore` / `LegacySSHStore`? Doing it here seems wrong (e.g. a binary cache URL like `http://[2001:db8::1]/` would be mangled to `http://2001:db8::1/`, which doesn't work with curl).

Reply:
Fair point, will take care of this.